app/src/main/java/com/concordium/wallet/ui/walletconnect/WalletConnectMessageFragment.kt

```diff
@@ -76,5 +76,11 @@ class WalletConnectMessageFragment : WalletConnectBaseFragment() {
         _viewModel.errorInt.observe(viewLifecycleOwner) {
             binding.sign.isEnabled = true
         }
+        _viewModel.isAuthenticationCanceled.observe(viewLifecycleOwner) {
```

Comment:
Wouldn’t it work just to remove the `isEnabled = false` so this

```
        binding.sign.setOnClickListener {
            binding.sign.isEnabled = false
            _viewModel.prepareMessage()
        }
```

become this

```
        binding.sign.setOnClickListener {
            _viewModel.prepareMessage()
        }
```

in file `WalletConnectMessageFragment`.

Reading through the code to me it doesn’t seem like we need to disable it just because it is clicked.